app/Api/Controllers/DashboardController.php

```diff
@@ -0,0 +1,190 @@
+                                                'size_x'       => $new_widget->size_x,
+                                                'title'        => $new_widget->title,
+                                                'refresh'      => $new_widget->refresh,
+                                                'settings'     => $new_widget->settings,
+                                                'dashboard_id' => $dashboard->dashboard_id,
+                                            ]);
+                    }
+                }
+                return $this->response->array(array('statusText' => 'OK', 'dashboard_id' => $dashboard->dashboard_id));
+            }
+            else {
+                return $this->response->errorInternal();
+            }
+        }
+        else {
+            $errors = $validation->errors();
+            return response()->json($errors,422);
+        }
+    }
+
+    /**
+     * Display the specified resource.
+     *
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function show(Request $request, $id)
+    {
+        $dashboard = Dashboard::find($id);
+        $widgets   = Dashboard::find($id)->widgets()->get();
+        // morph the data as required
+        if ($request->query('displayFormat') == 'human') {
+        }
+
+        return array('dashboard' => $dashboard, 'widgets' => $widgets);
+    }
+
+    /**
+     * Show the form for editing the specified resource.
+     *
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function edit($id)
+    {
+        //
+    }
+
+    /**
+     * Update the specified resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function update(Request $request, $id)
+    {
+        $validation = Validator::make($request->all(), [
+            'name' => 'required|max:255',
+            'access' => 'required',
+        ]);
+        if($validation->passes())
+        {
+            $dashboard = Dashboard::find($id);
+            $dashboard->dashboard_name = $request->name;
+            $dashboard->access         = $request->access;
+            if ($dashboard->save())
+            {
+                return $this->response->array(array('statusText' => 'OK'));
+            }
+            else {
+                return $this->response->errorInternal();
+            }
+        }
+        else {
+            $errors = $validation->errors();
+            return response()->json($errors,422);
+        }
+    }
+
+    /**
+     * Remove the specified resource from storage.
+     *
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function destroy($id)
+    {
+        if (Dashboard::destroy($id))
+        {
+            if (UsersWidgets::where('dashboard_id', $id)->delete() >= 0)
```

Comment:
Move this into the model.

http://laravel-tricks.com/tricks/cascading-deletes-with-model-events